content/docs/setup/kubernetes/spec-requirements/index.md

```diff
@@ -38,3 +38,67 @@ cluster must satisfy the following requirements:
+
+* _**Check required ports**_: The following table lists the default ports used by Istio components. Make sure the ports are available before installing Istio components.
+
+    | Component | Port | NodePort | Description |
+    |----|----|----|----|
+    | Citadel | 8060 |   | The port number for Citadel GRPC server |
+    | | | | |
+    | Core DNS | 53 |   | DNS - TCP/UDP Protocols |
+    | Core DNS | 8053 |   | DNS plugin |
+    | Core DNS | 9153 |   | DNS |
+    | | | | |
+    | Egress gateway | 80 |   |   |
+    | Egress gateway | 443 |   |   |
+    | Egress gateway | 15090 |   | Proxy |
+    | Egress gateway | 15443 |   | This is the port where SNI routing happens |
+    | | | | |
+    | Galley | 443 |   | Validation port |
+    | Galley | 9091 |   | Mesh control Plane - Running mixer |
+    | Galley | 9094 |   | Port to use for exposing profiling (no longer used?) |
+    | Galley | 15014 |   | Monitoring |
+    | | | | |
+    | Grafana | 3000 |   | Grafana |
+    | Grafana | 15031 |   | Grafana |
+    | | | | |
+    | Envoy | 15000 |   | Envoy admin port (commands/diagnostics) |
+    | | | | |
+    | Ingress gateway | 80 | 31380 |   |
+    | Ingress gateway | 443 | 31390 |   |
+    | Ingress gateway | 15029 |   | Kiali |
+    | Ingress gateway | 15030 |   | Prometheus |
+    | Ingress gateway | 15031 |   | Grafana |
+    | Ingress gateway | 15032 |   | Tracing |
+    | Ingress gateway | 15090 |   | Proxy |
+    | Ingress gateway | 15443 |   | This is the port where SNI routing happens |
+    | Ingress gateway | 31400 | 31400 | Sample port|
+    | | | | |
+    | Mixer | 9091 |   | Policy/Telemetry |
+    | Mixer | 15004 |   | Policy/Telemetry - `mTLS`|
+    | Mixer | 15014 |   | Policy/Telemetry monitoring port |
+    | Mixer | 15090 |   | Proxy |
+    | Mixer | 42422 |   | Telemetry - Prometheus |
+    | | | | |
+    | Pilot | 8080 |   | Pilot service - Discovery - legacy |
+    | Pilot | 15003 |   | Pilot service - Proxy - http - Discovery - old |
+    | Pilot | 15004 |   | Pilot service - Mixer |
+    | Pilot | 15005 |   | Pilot service - Proxy - https - Discovery |
+    | Pilot | 15007 |   | Pilot service - Proxy - http - Discovery |
+    | Pilot | 15010 |   | Pilot service - XDS pilot - discovery |
+    | Pilot | 15011 |   | Pilot service - `mTLS` - Proxy - discovery |
+    | Pilot | 15012 |   | Pilot service - `mTLS` - Proxy - Discovery service grpc address, with https discovery (still used?) |
+    | Pilot | 15014 |   | Pilot service - Monitoring |
+    | | | | |
+    | Prometheus | 9090 |   | Prometheus |
+    | | | | |
+    | Service Graph | 8088 |   | Service Graph |
+    | | | | |
+    | Sidecar injector | 443 |   | Sidecar injector |
+    | | | | |
+    | Tracing | 80 |   | Tracing - Jaeger |
+    | Tracing | 9411 |   | Tracing - Other |
+    | Tracing | 15032 |   | Tracing - Jaeger |
+    | Tracing | 16686 |   | Tracing - Jaeger |
+    | | | | |
+    | Zipkin | 9411 |   | Zipkin collector |
```

Comment:
Ok, I feel we also need to separate the ports to ports that user should care when moving their services to Istio which runs on the proxy mostly, and ports admin should care.  WDYT?

Reply:
I agree. This big table is not useful. There should just be a list entry in this document for the ports that pods can't use because of istio-proxy port conflict. As I commented earlier, if there is a table needed to list the purpose of these ports, it should be in a separate section of the document, and just referenced from the pod requirements list entry.

Other component ports that need to be free to deploy Istio should be documented separately and listed in a different table. Which ports have this problem and why? Clearly not all of them in this big list.

Reply:
I'll talk with @GregHanson again, but I think we have two sets of ports/protocols that services can't use: 1) Ports because they are already in use in the `Pod`, 2) Ports/Protocols because Pilot fails to push rules if there is a conflict. The later is a larger list and what this table describes. At least that is how I understand the problem and trying to document.